lib/cloudinary/uploader.rb

```diff
@@ -91,7 +91,7 @@ def self.upload_large(file, public_id_or_options={}, old_options={})
       public_id = public_id_or_options
       options   = old_options
     end
-    if file.match(REMOTE_URL_REGEX)
+    if file.is_a?(String) && file.match(REMOTE_URL_REGEX)
```

Comment:
I think it would be better to check `file.respond_to?(:match)` instead to allow duck-typing.
In addition, the same check should be added to https://github.com/cloudinary/cloudinary_gem/blob/23e94073719732701915c68a12ff68afaaa32650/lib/cloudinary/uploader.rb#L77